src/realm/parser/driver.cpp

```diff
@@ -1355,6 +1280,136 @@ std::unique_ptr<Subexpr> ConstantNode::visit(ParserDriver* drv, DataType hint)
     return ret;
 }
 
+std::vector<std::shared_ptr<Subexpr>> ConstantNode::clone_list_of_args(std::vector<Mixed>& mixed_args)
+{
+    std::vector<std::shared_ptr<Subexpr>> args_in_list;
```

Comment:
Instead of this, I'd recommend building up a `ConstantMixedList`. It should take care of owning the data. That will simplify a lot of code at the call site.

Reply:
If I understand correctly, 

```
 void ConstantMixedList ::set(size_t n, Mixed val)
    {
        Value<Mixed>::set(n, val);
        (*this)[n].use_buffer(m_buffer[n]);
    }
```

which copies the buffer only for string and binary.

```
switch (get_type()) {
        case type_String:
            buf = std::string(string_val);
            string_val = StringData(buf);
            break;
        case type_Binary:
            buf = std::string(binary_val);
            binary_val = BinaryData(buf);
            break;
        default:
            break;
    }
```

It seems to me that these are the only 2 types for which we need to deep copy stuff. So it should be fine. Thanks for the suggestion.